src/irmin-pack/unix/dispatcher.ml

```diff
@@ -250,4 +250,80 @@ module Make (Fm : File_manager.S with module Io = Io.Unix) :
   let shrink_accessor_exn a ~new_len =
     if new_len > a.len then failwith "shrink_accessor_exn to larger accessor";
     { a with len = new_len }
+
+  let create_sequential_accessor_exn location rem_len ~poff ~len =
+    if len > rem_len then raise (Errors.Pack_error `Read_out_of_bounds)
+    else { poff; len; location }
+
+  let create_sequential_accessor_from_range_exn location rem_len ~poff ~min_len
+      ~max_len =
+    let len =
+      if rem_len < min_len then raise (Errors.Pack_error `Read_out_of_bounds)
+      else if rem_len > max_len then max_len
+      else rem_len
+    in
+    { poff; len; location }
+
+  let create_sequential_accessor_seq t ~min_header_len ~max_header_len ~read_len
+      =
+    let chunks =
+      match Fm.mapping t.fm with
+      | Some mapping ->
+          let chunks = ref [] in
+          Mapping_file.iter mapping (fun ~off ~len ->
+              chunks := (off, len) :: !chunks)
+          |> Errs.raise_if_error;
+          List.rev !chunks
+      | None -> []
+    in
+    let suffix_end_offset = Fm.Suffix.end_offset (Fm.suffix t.fm) in
+    let entry_offset_suffix_start = entry_offset_suffix_start t in
+    let buf = Bytes.create max_header_len in
```

Comment:
I haven't thought about the full implications here, but since we use `Bytes.unsafe_to_string` quite a bit in the code, I think we ought to at least consider if using a shared buffer here is appropriate.

Reply:
This byte is a bit different from the others, as we know from the start which size will always fit (max_header_len, which is a small size as well).

We could consider using the same `Bytes.t` as other places, but I do not know if we will really gain much from it

Reply:
Ah, my bad. My last sentence is ambiguous. Let me try again. :sweat_smile: 

`buf` here is reused across multiple invocations of (now) `get_entry_accessor` so it is shared. It likely is not a problem with how you want to use it (if `read_len` promises to not do anything with `Bytes.unsafe_to_string` and store/use the results after its initial invocation), but it would (I think) be safer to create a new `buf` for each invocation of `get_entry_accessor`.

Reply:
I see, a better solution would be to give `read_len` a string and use `Bytes.unsafe_to_string` ourselves. It is what was planned at the beginning when @Ngoguey42 and me talked about it, but I forgot to do it.

This way we have no issues with the sub-function modifying it (except if they do some shenanigans with `unsafe_of_string`, which is highly unlikely to happen) and we do not recreate the buffer every time.

Is it a good compromise or should we really go for an initialization every times ?

Reply:
So something like?

```ocaml
let str = Bytes.unsafe_to_string buf in
read_len str
```

The issue with this is if `read_len` stores `str` and tries to use it later (I have no reason to think it would but merely thinking through possibilities). Once `buf` is overwritten by a subsequent accessor, the original `str` will be different since it is using `buf` as its storage.

An excerpt from [docs on `unsafe_to_string`](https://v2.ocaml.org/api/Bytes.html#unsafe):

> Unique ownership is linear: passing the data to another piece of code means giving up ownership (we cannot write the data again). A unique owner may decide to make the data shared (giving up mutation rights on it), but shared data may not become uniquely-owned again.

Without in-depth thinking, I do like handing `read_len` a string. If we make that change, I see two options to be safe:

1. Do not re-use `buf` and use `Bytes.unsafe_to_string` (unique ownership is passed along)
2. Re-use `buf` and use `Bytes.to_string` (no need to think about unsafe operations)

Reply:
Good point, I would go for the first option